internal/daemon/worker/worker.go

```diff
@@ -186,7 +186,7 @@ func (w *Worker) Start() error {
 		return err
 	}
 
-	nodeCreds, err := types.NewNodeCredentials(w.baseContext, w.NodeeFileStorage)
+	nodeCreds, err := types.NewNodeCredentials(w.baseContext, w.NodeeFileStorage, nodee.WithWrapper(w.conf.WorkerStorageKms))
```

Comment:
should this be moved up to `New(conf *Config)`?  I'm curious is there any reason to wait until startup to setup these creds?  Also, if we move it up to `New(...)` we could easily write a unit test for this using a "real" worker vs a TestWorker.

Reply:
This is a good question. I'm guessing it's because the nodee functions need a context and there already is one in `Start()` but none in `New()`. Let's ask @jefferai what he thinks about moving them, since I believe he put them there in the llb-byow base branch

Reply:
not sure there's anything magical about the `context.WithCancel()` that's created 20 lines above in this same func.  I'm pretty sure there's no way to call `cancel` on that context before this func returns.  As I stated before, moving it to `New(...)` has the extra added bonus of being able to easily create a unit test for it... which in my mind is quite compeling if there's no concrete reason to keep it here.